rfd/0001-testing-guidelines.md

```diff
@@ -117,21 +117,21 @@ Subtests
 ```go
 func TestParseInt(t *testing.T) {
 	tests := []struct {
-		desc    string
-		in      string
-		want    int
-		wantErr bool
+		desc      string
+		in        string
+		want      int
+		assertErr require.ErrorAssertionFunc
 	}{
-		{desc: "positive", in: "123", want: 123},
-		{desc: "negative", in: "-123", want: -123},
-		{desc: "non-numeric", in: "abc", wantErr: true},
-		{desc: "empty", in: "", wantErr: true},
+		{desc: "positive", in: "123", want: 123, assertErr: require.NoError},
+		{desc: "negative", in: "-123", want: -123, assertErr: require.NoError},
+		{desc: "non-numeric", in: "abc", assertErr: require.Error},
+		{desc: "empty", in: "", assertErr: require.Error},
```

Comment:
> With require, error is checked first and if that check fails, want is never checked.

Sorry andrew, i'm still confused.

`assertErr: require.Error`, this means we are asserting that it returned an error (not nil), so this is a check that passes. Doesn't that mean the `require.Equal(t, got, tt.want)` will still run? (meaning we still need want values)

Reply:
ah, right, it still runs but we assume that the function under test returns the zero value of `want` (0)

Reply:
ah i get it now! i forgot about the zero values if not explicitly defined.